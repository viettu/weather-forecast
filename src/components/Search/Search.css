.Search-form .Search-control {
    display: block;
    flex: 1;
    margin-right: 10px;
}
