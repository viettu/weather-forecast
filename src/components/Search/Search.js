import React, {useState} from 'react';
import {Form, Button, Container, Row} from 'react-bootstrap';
import PropTypes from 'prop-types';

import './Search.css';

const Search = ({onSearch, placeholder, autoFocus}) => {
    const [term, setTerm] = useState('');

    const handleSubmit = (event) => {
        event.preventDefault();

        if (onSearch) {
            onSearch(term);
        }
    };

    return (
        <Form onSubmit={handleSubmit} className="Search-form" data-testid="search-form">
            <Container>
                <Row>
                    <Form.Control
                        className="Search-control"
                        type="text"
                        autoFocus={autoFocus}
                        placeholder={placeholder}
                        value={term}
                        onChange={(e) => setTerm(e.target.value)}
                    ></Form.Control>
                    <Button variant="primary" type="submit">
                        Search
                    </Button>
                </Row>
            </Container>
        </Form>
    );
};

Search.propTypes = {
    onSearch: PropTypes.func,
    placeholder: PropTypes.string,
    autoFocus: PropTypes.bool,
};
Search.defaultProps = {
    placeholder: 'Search',
};

export default Search;
