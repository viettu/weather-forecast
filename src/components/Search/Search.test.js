import React from 'react';
import {render, screen, cleanup, fireEvent} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import Search from './Search';

afterEach(() => {
    cleanup();
});

test('renders without crashing', () => {
    render(<Search />);
});

test('snapshot testing', () => {
    let renderResult = render(<Search />);
    expect(renderResult.container.firstChild).toMatchSnapshot();

    renderResult = render(<Search placeholder="Custom placeholder" />);
    expect(renderResult.container.firstChild).toMatchSnapshot();
});

test('should display default placeholder', () => {
    const {queryByPlaceholderText} = render(<Search />);
    expect(queryByPlaceholderText('Search')).toBeVisible();
});

test('should display placeholder', () => {
    const {queryByPlaceholderText} = render(<Search placeholder="Enter a city" />);
    expect(queryByPlaceholderText('Enter a city')).toBeVisible();
});

test('should trigger onSearch callback', () => {
    const handleSearch = jest.fn();
    render(<Search onSearch={handleSearch} />);

    const searchFrom = screen.getByTestId('search-form');
    fireEvent.submit(searchFrom);

    expect(handleSearch).toHaveBeenCalledTimes(1);
});

test('click search button should trigger onSearch callback', () => {
    const handleSearch = jest.fn();
    render(<Search onSearch={handleSearch} />);

    const searchButton = screen.getByText('Search', {selector: 'button'});
    userEvent.click(searchButton);

    expect(handleSearch).toHaveBeenCalledTimes(1);
});
