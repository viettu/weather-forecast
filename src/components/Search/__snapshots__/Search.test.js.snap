// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<form
  class="Search-form"
  data-testid="search-form"
>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <input
        class="Search-control form-control"
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        class="btn btn-primary"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
</form>
`;

exports[`snapshot testing 2`] = `
<form
  class="Search-form"
  data-testid="search-form"
>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <input
        class="Search-control form-control"
        placeholder="Custom placeholder"
        type="text"
        value=""
      />
      <button
        class="btn btn-primary"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
</form>
`;
